### ISSUE TITLE
TensorFlow build pip package fails while using pyenv local OR <some script> fails because python fails to load modules or gives syntax errors

### ISSUE THREAD

**Issue by wadkar** [yyuu/pyenv#601]
#### TL;DR
If you experience failure while executing a script that issues `python` commands or executes another python script:
 - try executing the command again in the appropriate `pyenv shell`
 - check if the command is a python script or invokes a python script and fix the shebang to `#!/usr/bin/env python`
 
In my limited experience, such failures usually show up as:
  - incompatible python version, but you have the right version installed
  - module not found but you know you have that module installed
  - something that shows up as simple as `pip install foo` on stackoverflow
  - the issue can be traced back to something related to `PYTHONPATH`

#### For the sake of posterity
This is more for the sake of posterity for the rare soul who is trying to compile [TensorFlow](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/g3doc/get_started/os_setup.md#create-the-pip-package-and-install) and build a pip package for TensorFlow. If you rely on `pyenv local` and `.python-version` to manage your python environment, then your `build_pip_package` command might fail as follows:

```sh
# bazel-bin/tensorflow/tools/pip_package/build_pip_package /tmp/tensorflow_pkg
Sun May 15 02:25:10 EDT 2016 : === Using tmpdir: /tmp/tmp.65EN3pAdGL
/tmp/tmp.65EN3pAdGL ~/git/tensorflow/tensorflow
Sun May 15 02:25:10 EDT 2016 : === Building wheel
Traceback (most recent call last):
  File "setup.py", line 26, in <module>
    from setuptools import find_packages, setup, Command, Extension
ImportError: No module named setuptools
```
The script that builds the pip package does so in a temp directory. If you rely on `pyenv local` and generate `.python-version` file to tell pyenv which python to pick, then it won't work as the temp directory won't have the `.python-version` file. In this case `pyenv shell` works just fine.

On a totally unrelated note, scripts that start with a shebang `#!/usr/bin/python` would fail even if run within `pyenv shell`. Either run that script as `python scriptName` or fix the shebang to `#!/usr/bin/env python` and execute in a `pyenv shell` for a more reliable python environment execution.

**Comment by wadkar** [comment]
Now that I think about it, the README/wiki is much better place for this. Feel free to include it in the README.md or wiki.

**Comment by wadkar** [comment]
I just updated the FAQ wiki page with the TL;DR piece. Hope it's OK and helps somebody (probably me 😄) in future.

**Comment by samvv** [comment]
Thanks for this, I was having exactly the same problem while building TensorFlow. In my case, though, the only thing that worked was setting `pyenv global`.

**Comment by wadkar** [comment]
@ghost Maybe add that to the wiki too!
